src/react/HomePage.tsx

```diff
@@ -19,111 +19,66 @@
-      const latestMajor = Math.max(...interfaceMajors);
-      navigate(`/interfaces/${interfaceName}/${latestMajor}`);
-    });
-  }, []);
+const ServiceStatusRow = ({ service, status }: ServiceStatusRowProps): React.ReactElement => {
+  let messageCell;
 
-  const cellSpacingClass = 'mb-3';
+  if (status === 'loading') {
+    messageCell = (
+      <td>
+        <Spinner animation="border" role="status" />
+      </td>
+    );
+  } else if (status === 'ok') {
+    messageCell = (
+      <td className="color-green">
+        <i className="fas fa-check-circle mr-1" />
+        This service is operating normally
+      </td>
+    );
+  } else {
+    messageCell = (
+      <td className="color-red">
+        <i className="fas fa-times-circle mr-1" />
+        This service appears offline
+      </td>
+    );
+  }
 
   return (
-    <Container fluid className="p-3">
-      <Row>
-        <Col xs={12}>
-          <h2 className="mb-4">Astarte Dashboard</h2>
-        </Col>
-        <Col xs={6} className={cellSpacingClass}>
-          <ApiStatusCard
-            appengine={appEngineHealth.status}
-            realmManagement={realmManagementHealth.status}
-            pairing={pairingHealth.status}
-            showFlowStatus={astarte.config.enableFlowPreview}
-            flow={astarte.config.enableFlowPreview ? flowHealth.status : null}
-          />
-        </Col>
-        <WaitForData data={devicesStats.value} status={devicesStats.status}>
-          {({ connected_devices: connectedDevices, total_devices: totalDevices }) => (
-            <Col xs={6} className={cellSpacingClass}>
-              <DevicesCard
-                connectedDevices={connectedDevices}
-                totalDevices={totalDevices}
-                connectedDevicesProvider={connectedDevicesProvider}
-              />
-            </Col>
-          )}
-        </WaitForData>
-        <WaitForData data={interfaces.value} status={interfaces.status}>
-          {(interfaceList) => (
-            <Col xs={6} className={cellSpacingClass}>
-              <InterfacesCard
-                interfaceList={interfaceList}
-                onInterfaceClick={redirectToLastInterface}
-                onInstallInterfaceClick={() => navigate('/interfaces/new')}
-              />
-            </Col>
-          )}
-        </WaitForData>
-        <WaitForData data={triggers.value} status={triggers.status}>
-          {(triggerList) => (
-            <Col xs={6} className={cellSpacingClass}>
-              <TriggersCard
-                triggerList={triggerList}
-                onInstallTriggerClick={() => navigate('/triggers/new')}
-              />
-            </Col>
-          )}
-        </WaitForData>
-      </Row>
-    </Container>
+    <tr>
+      <td>{service}</td>
+      {messageCell}
+    </tr>
   );
 };
 
-const ApiStatusCard = ({ appengine, realmManagement, pairing, showFlowStatus, flow }) => (
+interface ApiStatusCardProps {
+  appengine: 'loading' | 'ok' | 'err';
```

Comment:
this type comes from our fetch hook. We should probably define a type and import it here

Reply:
I'd rather not couple the types here with the types of the fetch hook. However I will define a type to reuse within this file